From 92ece85d4bb2760c0ae40500938be25bd7dcd72a Mon Sep 17 00:00:00 2001
From: Chris Olekas <chris.olekas@ssimwave.com>
Date: Wed, 20 Aug 2025 11:03:30 -0400
Subject: [PATCH] libavfilter/af_join - fix memory leak

---
 libavfilter/af_join.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libavfilter/af_join.c b/libavfilter/af_join.c
index 0ea53248b6..bd197d3edc 100644
--- a/libavfilter/af_join.c
+++ b/libavfilter/af_join.c
@@ -195,7 +195,9 @@ static av_cold void join_uninit(AVFilterContext *ctx)
     for (i = 0; i < s->inputs && s->input_frames; i++) {
         av_frame_free(&s->input_frames[i]);
     }
-
+    if (s->ch_layout.u.map && (s->ch_layout.order == AV_CHANNEL_ORDER_CUSTOM)) {
+        av_freep(&s->ch_layout.u.map);
+    }
     av_freep(&s->channels);
     av_freep(&s->buffers);
     av_freep(&s->input_frames);
-- 
2.43.0

